src/win/fs.c

```
@@ -1971,8 +1971,51 @@ INLINE static DWORD fs__stat_impl_from_path(WCHAR* path,
                        flags,
                        NULL);
 
-  if (handle == INVALID_HANDLE_VALUE)
-    return GetLastError();
+  if (handle == INVALID_HANDLE_VALUE) {
+    DWORD error = GetLastError();
+    int sys_error = uv_translate_sys_error(error);
+    WIN32_FIND_DATAW findFile_data;
+    HANDLE find_handle;
+    FILE_STAT_BASIC_INFORMATION stat_info;
+
+    if (sys_error != UV_EBUSY && sys_error != UV_EPERM) {
+      return error;
+    }
+
+    // If the file isn't open, try to get the file statistics using FindFirstFileW
+    find_handle = FindFirstFileW(path, &findFile_data);
+
+    if (find_handle == INVALID_HANDLE_VALUE) {
+      return GetLastError();
```

Should we return this error or the original one?